WIP enable aot

inforithmics opened the pull request:
Enables AOT analyzing, does not work yet because XMLSerializer is not AOT compatible
Maybe XmlSerializer Generator helps https://learn.microsoft.com/en-us/dotnet/core/additional-tools/xml-serializer-generator

pauldendulk commented:
Microsoft.XmlSerializer.Generator is a nuget package. I would not want to add a dependency to the BruTile nuget package for this, but I guess this package is only acting during building/compiling and it is not needed for the BruTile nuget. Is that correct?

inforithmics commented:
Yes this is correct. The nuget needs to be marked as private so that it isn’t included in the dependencies.

inforithmics commented:
Things done: 
1. I added Targets netstandard2.1 (Because netstandard2.0 does not support Microsofot.XmlSerializer.Generator)
2. I added .NET 7.0 and .NET 8.0 Targets because they support isAOTCompatbile
3. updated sqlite dependencies in Brutile.Mbtiles because only newest Sqlite is compatible with Target changes in .NET 8

pauldendulk commented on Directory.Build.props:
```diff
@@ -13,14 +13,25 @@
 		<Company>BruTile Developers Team</Company>
 		<LangVersion>10.0</LangVersion>
-		<TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+		<TreatWarningsAsErrors>false</TreatWarningsAsErrors>
```
I would like to keep TreatWarningsAsErrors true. It saves a lot of trouble in the long run. Not sure what you are trying to suppress. Could a global suppression be added that applies to everything regarding this one specific rule?

inforithmics replied:
This wasn‘t intentional I had to disable treat warnings as errors so that I could suppress the warning. Will restore the warnings as errors.

pauldendulk reviewed (commented):
Thanks a lot! 

I have a question about the TreatWarningsAsErrors is false.  See remark.

Another question is whether we should bump the version patch, minor or major. The minimum package version is now 1.8.116. If user use 1.6.292 in their app they would get a compile error. I guess if you follow the rules it should be a major bump. Silly because there are otherwise no changes. Still I think we should do it. It is just communication.

inforithmics commented on BruTile.MbTiles/BruTile.MbTiles.csproj:
```diff
@@ -12,6 +12,7 @@
 	<ItemGroup>
-		<PackageReference Include="sqlite-net-pcl" Version="1.6.292" />
+		<PackageReference Include="sqlite-net-pcl" Version="1.8.116" />
+		<PackageReference Include="SQLitePCLRaw.bundle_green" Version="2.1.7" />
```
SQLite-net-pcl references an older bundle green that gives some targeting warnings in .net 8 and Linux. So I could remove this to simplify the dependencies.

inforithmics replied:
This is the warning and in newer bundle_green it is fixed
`Warning: C:\Program Files\dotnet\sdk\8.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(284,5): warning NETSDK1206: Found version-specific or distribution-specific runtime identifier(s): alpine-x64. Affected libraries: SQLitePCLRaw.lib.e_sqlite3. In .NET 8.0 and higher, assets for version-specific and distribution-specific runtime identifiers will not be found by default. See https://aka.ms/dotnet/rid-usage for details. [D:\a\BruTile\BruTile\BruTile.MbTiles\BruTile.MbTiles.csproj::TargetFramework=net8.0]`
But for simplicity I remove the QLitePCLRaw.bundle_green  for now

inforithmics commented on BruTile/BruTile.csproj:
```diff
@@ -1,11 +1,30 @@
 	</PropertyGroup>
 
+	<ItemGroup Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netstandard2.1'))">
+		<PackageReference Include="Microsoft.XmlSerializer.Generator" Version="8.0.0">
```
It adds Microsoft.XmlSerializer.Generator greater equal than netstandard2.1 which means
netstandard2.1
Net6
Net7
Net8
I did this because netstandrad2.0 is not supported by Microsoft.XmlSerializer.Generator and with netstandard2.1 even Xamarin.Forms on Android and iOS is supported

pauldendulk replied:
I see, I misunderstood IsTargetFrameworkCompatible. It is mentioned just twice in the entire Microsoft docs.
- https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/8.0/trimming-unsupported-targetframework
- https://learn.microsoft.com/en-us/dotnet/core/tools/sdk-errors/

I considered dropping netstandard2.0 but that would mean dropping support for .NET framework which I don't want to do right now. If netstandard2.0 is dropped I think we also should drop netstandard2.1 and make .net6 the lowest supported version.
